kamelet-support/src/main/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteFileProcessor.java

```diff
@@ -0,0 +1,315 @@
+                            Parameter>> typeToClassConversion = Map.of(
+                            "string", this::getStringParameter,
+                            "object", this::getObjectParameter,
+                            "integer", this::getNumberParameter,
+                            "boolean", this::getBooleanParameter
+                    );
+
+                    Parameter parsedParameter = typeToClassConversion
+                            .getOrDefault(parameterData.get("type"),
+                                    this::getObjectParameter)
+                            .apply(parameter);
+
+                    if ("path".equals(parameterData.get("kind"))) {
+                        parsedParameter.setPath(true);
+                    }
+
+                    return parsedParameter;
+                }
+        ).toList();
+
+        return new JsonCamelObject(
+                id,
+                camelKind,
+                defaultIcon,
+                title,
+                description,
+                defaultGroup,
+                type,
+                parameters,
+                requiredProperties
+        );
+    }
+
+    private StringParameter getStringParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        StringParameter parsedParameter = new StringParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null),
+                null
+        );
+
+        return parsedParameter;
+    }
+
+    private NumberParameter getNumberParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        //IntegerParameter doesn't fit some long types there are present
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        NumberParameter parsedParameter = new NumberParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                Long.parseLong(parameterData
+                        .getOrDefault("defaultValue", "0")
+                        .replaceAll("L", ""))
+        );
+
+        return parsedParameter;
+    }
+
+    private ObjectParameter getObjectParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        ObjectParameter parsedParameter = new ObjectParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null)
+        );
+
+        return parsedParameter;
+    }
```

Comment:
I thought that in this case there was some value to explicitly tell that we are returning the parsed parameter, but you are right, it can be summarized